models/stg_iterable__event.sql

```diff
@@ -28,27 +28,33 @@ fields as (
 
 final as (
     
-    select 
-        _fivetran_id as event_id,
-        campaign_id,
-        content_id,
+    select
+        cast(_fivetran_id as {{ dbt.type_string() }} ) as event_id,
+        cast(campaign_id as {{ dbt.type_string() }} ) as campaign_id,
+        cast(content_id as {{ dbt.type_string() }} ) as content_id,
         created_at,
         cast( {{ dbt.date_trunc('day', 'created_at') }} as date) as created_on,
         lower(email) as email,
         additional_properties,
         event_name,
-        message_bus_id,
-        message_id,
-        message_type_id,
+        cast(message_bus_id as {{ dbt.type_string() }} ) as message_bus_id,
+        cast(message_id as {{ dbt.type_string() }} ) as message_id,
+        cast(message_type_id as {{ dbt.type_string() }} ) as message_type_id,
         recipient_state,
         status,
         transactional_data,
         unsub_source,
         user_agent,
         user_agent_device,
-        _fivetran_synced
+        _fivetran_synced,
+        cast(_fivetran_user_id as {{ dbt.type_string() }} ) as _fivetran_user_id
+
+        {{ fivetran_utils.fill_pass_through_columns('iterable_event_pass_through_columns') }}
+
     from fields
 )
 
-select * 
+select 
+    *,
+    coalesce(_fivetran_user_id, email) as unique_user_key
```

Comment:
Ahh. I remember the other reason now. `_fivetran_user_id` is read in as an integer while `email` is a string. So I did it in the final select statement in order for the casts to take place in the prior CTE. 

I updated it to cast as string in place, for all the models, to prevent any issues with data types